File: app/src/main/java/com/exceptionteam17/bestcookingconverter/fragments/TempFragment.java
package com.exceptionteam17.bestcookingconverter.fragments;

import android.annotation.SuppressLint;
import android.graphics.PorterDuff;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.core.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.TextView;

import com.exceptionteam17.bestcookingconverter.R;

import java.text.DecimalFormat;
import java.text.NumberFormat;

public class TempFragment extends Fragment implements View.OnClickListener{

    private View view;
    private EditText editFrom, editTo;
    private TextView text1, text2;
    private boolean isCels;
    private Button btn_1, btn_2, btn_3, btn_4, btn_5, btn_6, btn_7, btn_8, btn_9, btn_0, btn_dot, btn_del, btn_clear;
    private ImageButton arow;


    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        view = inflater.inflate(R.layout.convertor_temp_layout, container, false);
        initialise();
        calculateAll();
        return view;
    }

    private void initialise() {
        isCels = true;
        editFrom = view.findViewById(R.id.con_edit_1);
        editFrom.setText("0");
        editFrom.getBackground().setColorFilter(ContextCompat.getColor(view.getContext(), R.color.textColor), PorterDuff.Mode.SRC_ATOP);
        editTo = view.findViewById(R.id.con_edit_2);
        editTo.setVisibility(View.VISIBLE);
        editTo.setText("0");
        editTo.getBackground().setColorFilter(ContextCompat.getColor(view.getContext(), R.color.textColor), PorterDuff.Mode.SRC_ATOP);

        btn_1 = view.findViewById(R.id.btn_1);
        btn_1.setOnClickListener(this);
        btn_2 = view.findViewById(R.id.btn_2);
        btn_2.setOnClickListener(this);
        btn_3 = view.findViewById(R.id.btn_3);
        btn_3.setOnClickListener(this);
        btn_4 = view.findViewById(R.id.btn_4);
        btn_4.setOnClickListener(this);
        btn_5 = view.findViewById(R.id.btn_5);
        btn_5.setOnClickListener(this);
        btn_6 = view.findViewById(R.id.btn_6);
        btn_6.setOnClickListener(this);
        btn_7 = view.findViewById(R.id.btn_7);
        btn_7.setOnClickListener(this);
        btn_8 = view.findViewById(R.id.btn_8);
        btn_8.setOnClickListener(this);
        btn_9 = view.findViewById(R.id.btn_9);
        btn_9.setOnClickListener(this);
        btn_0 = view.findViewById(R.id.btn_0);
        btn_0.setOnClickListener(this);
        btn_dot = view.findViewById(R.id.btn_dot);
        btn_dot.setOnClickListener(this);
        btn_del = view.findViewById(R.id.btn_del);
        btn_del.setOnClickListener(this);
        btn_clear = view.findViewById(R.id.btn_clear);
        btn_clear.setOnClickListener(this);

        text1 = view.findViewById(R.id.text_temp_1);
        text1.setText(R.string.cel);
        text2 = view.findViewById(R.id.text_temp_2);
        text2.setText(R.string.far);

        arow = view.findViewById(R.id.arow);
        arow.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                isCels = !isCels;
                if(isCels){
                    text1.setText(R.string.cel);
                    text2.setText(R.string.far);
                } else {
                    text2.setText(R.string.cel);
                    text1.setText(R.string.far);
                }
                calculateAll();
            }
        });
    }


    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.btn_1:
                addNumber(1);
                calculateAll();
                break;
            case R.id.btn_2:
                addNumber(2);
                calculateAll();
                break;
            case R.id.btn_3:
                addNumber(3);
                calculateAll();
                break;
            case R.id.btn_4:
                addNumber(4);
                calculateAll();
                break;
            case R.id.btn_5:
                addNumber(5);
                calculateAll();
                break;
            case R.id.btn_6:
                addNumber(6);
                calculateAll();
                break;
            case R.id.btn_7:
                addNumber(7);
                calculateAll();
                break;
            case R.id.btn_8:
                addNumber(8);
                calculateAll();
                break;
            case R.id.btn_9:
                addNumber(9);
                calculateAll();
                break;
            case R.id.btn_0:
                addNumber(0);
                calculateAll();
                break;
            case R.id.btn_dot:
                addDot();
                break;
            case R.id.btn_del:
                deleteOneChar();
                calculateAll();
                break;
            case R.id.btn_clear:
                clear();
                calculateAll();
                break;
        }
    }

    @SuppressLint("SetTextI18n")
    private void addNumber(int i) {

        String text = editFrom.getText().toString().trim();

        if(editFrom.getText().toString().trim().matches("[0-9]{6}+")){
            return;
        }

        if (editFrom.getText().toString().trim().equalsIgnoreCase("0")) {
            if(i == 0) {
                return;
            } else {
                editFrom.setText("");
            }
        }

        if (editFrom.getText().toString().trim().length() < 10) {
            if (!(text.length() == 1 && text.equals("0") && i == 0)) {
                if (!editFrom.getText().toString().trim().matches("[0-9]\\.[0-9]{3}")) {
                    editFrom.setText(editFrom.getText().toString() + i);
                }
            }
        }
    }


    @SuppressLint("SetTextI18n")
    private void addDot() {
        String text = editFrom.getText().toString();
        int countDots = 0;
        for (int i = 0; i < text.length(); i++) {
            if (text.charAt(i) == '.') {
                countDots++;
            }
        }

        if (countDots == 0) {
            if (editFrom.getText().toString().trim().length() < 11) {
                editFrom.setText(editFrom.getText().toString() + ".");
            }
        }
    }

    private void deleteOneChar() {
        String s = editFrom.getText().toString();
        if (s.length() > 1) {
            editFrom.setText(s.substring(0, s.length()-1));
        } else {
            editFrom.setText("0");
        }
    }

    private void clear(){
        editFrom.setText("0");
    }

    private void calculateAll(){
        double valInGr =isCels?
                ((Double.parseDouble(editFrom.getText().toString()) * 9) / 5) + 32 :
                ((Double.parseDouble(editFrom.getText().toString()) - 32) * 5) / 9;

        NumberFormat formatter = new DecimalFormat("#.###");
        String temp = formatter.format(valInGr);
        editTo.setText(temp);
    }
}

File: app/src/main/java/com/exceptionteam17/bestcookingconverter/fragments/TempFragment.kt
package com.exceptionteam17.bestcookingconverter.fragments

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.exceptionteam17.bestcookingconverter.MainActivity.Companion.buttonClickAnim
import com.exceptionteam17.bestcookingconverter.R
import kotlinx.android.synthetic.main.convertor_temp_layout.*
import java.text.DecimalFormat
import java.text.NumberFormat

class TempFragment : Fragment(), View.OnClickListener {
    private var myView: View? = null
    private var editFrom: TextView? = null
    private var editTo: TextView? = null
    private var isCels = false

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        myView = inflater.inflate(R.layout.convertor_temp_layout, container, false)
        return myView!!
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initialise()
        calculateAll()
    }

    private fun initialise() {
        isCels = true
        editFrom = myView!!.findViewById(R.id.con_edit_1)
        editFrom!!.setText("0")
        //        editFrom.getBackground().setColorFilter(ContextCompat.getColor(view.getContext(), R.color.textColor), PorterDuff.Mode.SRC_ATOP);
        editTo = myView!!.findViewById(R.id.con_edit_2)
        editTo!!.visibility = View.VISIBLE
        editTo!!.setText("0")
        //        editTo.getBackground().setColorFilter(ContextCompat.getColor(view.getContext(), R.color.textColor), PorterDuff.Mode.SRC_ATOP);
        btn_1.setOnClickListener(this)
        btn_2.setOnClickListener(this)
        btn_3.setOnClickListener(this)
        btn_4.setOnClickListener(this)
        btn_5.setOnClickListener(this)
        btn_6.setOnClickListener(this)
        btn_7.setOnClickListener(this)
        btn_8.setOnClickListener(this)
        btn_9.setOnClickListener(this)
        btn_0.setOnClickListener(this)
        btn_dot.setOnClickListener(this)
        btn_del.setOnClickListener(this)
        btn_clear.setOnClickListener(this)
        text_temp_1.setText(R.string.cel)
        text_temp_2.setText(R.string.far)
        arow.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        v.startAnimation(buttonClickAnim)
        when (v.id) {
            R.id.btn_1 -> {
                addNumber(1)
                calculateAll()
            }
            R.id.btn_2 -> {
                addNumber(2)
                calculateAll()
            }
            R.id.btn_3 -> {
                addNumber(3)
                calculateAll()
            }
            R.id.btn_4 -> {
                addNumber(4)
                calculateAll()
            }
            R.id.btn_5 -> {
                addNumber(5)
                calculateAll()
            }
            R.id.btn_6 -> {
                addNumber(6)
                calculateAll()
            }
            R.id.btn_7 -> {
                addNumber(7)
                calculateAll()
            }
            R.id.btn_8 -> {
                addNumber(8)
                calculateAll()
            }
            R.id.btn_9 -> {
                addNumber(9)
                calculateAll()
            }
            R.id.btn_0 -> {
                addNumber(0)
                calculateAll()
            }
            R.id.btn_dot -> addDot()
            R.id.btn_del -> {
                deleteOneChar()
                calculateAll()
            }
            R.id.btn_clear -> {
                clear()
                calculateAll()
            }
            R.id.arow -> {
                isCels = !isCels
                if (isCels) {
                    text_temp_1.setText(R.string.cel)
                    text_temp_2.setText(R.string.far)
                } else {
                    text_temp_2.setText(R.string.cel)
                    text_temp_1.setText(R.string.far)
                }
                calculateAll()
            }
        }
    }

    @SuppressLint("SetTextI18n")
    private fun addNumber(i: Int) {
        val text = editFrom!!.text.toString().trim { it <= ' ' }
        if (editFrom!!.text.toString().trim { it <= ' ' }.matches("[0-9]{6}+".toRegex())) {
            return
        }
        if (editFrom!!.text.toString().trim { it <= ' ' }.equals("0", ignoreCase = true)) {
            if (i == 0) {
                return
            } else {
                editFrom!!.text = ""
            }
        }
        if (editFrom!!.text.toString().trim { it <= ' ' }.length < 10) {
            if (!(text.length == 1 && text == "0" && i == 0)) {
                if (!editFrom!!.text.toString().trim { it <= ' ' }.matches("[0-9]\\.[0-9]{3}".toRegex())) {
                    editFrom!!.text = editFrom!!.text.toString() + i
                }
            }
        }
    }

    @SuppressLint("SetTextI18n")
    private fun addDot() {
        val text = editFrom!!.text.toString()
        var countDots = 0
        for (i in 0 until text.length) {
            if (text[i] == '.') {
                countDots++
            }
        }
        if (countDots == 0 && editFrom!!.text.toString().trim { it <= ' ' }.length < 11) {
            editFrom!!.text = editFrom!!.text.toString() + "."
        }
    }

    private fun deleteOneChar() {
        val s = editFrom!!.text.toString()
        if (s.length > 1) {
            editFrom!!.text = s.substring(0, s.length - 1)
        } else {
            editFrom!!.text = "0"
        }
    }

    private fun clear() {
        editFrom!!.text = "0"
    }

    private fun calculateAll() {
        val valInGr = if (isCels) editFrom!!.text.toString().toDouble() * 9 / 5 + 32 else (editFrom!!.text.toString().toDouble() - 32) * 5 / 9
        val formatter: NumberFormat = DecimalFormat("#.###")
        val temp = formatter.format(valInGr)
        editTo!!.text = temp
    }
}